### PR TITLE
[Solidity] Bug fixes for Inheritance 

### PR DESCRIPTION
**Pull request by ChenfengWei0**
This PR fixed the known issues in Solidity inheritance, including missing function calls in the entry function, incorrect variable names, the order of constructor calls, et al.

### CODE REVIEW

**Comment by ChenfengWei0** [comment]
I don't understand why I failed  the `
PR / Check C/C++ code-style (pull_request) ` test. Could someone please help me with this?

```sh
fatal: invalid reference: inheritance_fix
Error: Invalid status code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v5/index.js:17:19)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5) {
  code: 128
}
Error: Invalid status code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v5/index.js:17:19)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
```

**Review by lucasccordeiro — APPROVED** [review]
Thanks for submitting this PR, @ChenfengWei0.

**Comment by brcfarias** [comment]
> I don't understand why I failed the `PR / Check C/C++ code-style (pull_request)` test. Could someone please help me with this?

I think this is happening because the CI is trying to add a commit to fix the indentation in the 'inheritance_fix' branch in this repository, but the branch only exists in your fork (ChenfengWei0).

**Comment by lucasccordeiro** [comment]
@ChenfengWei0: you can use clang-format to fix this issue on your branch.

**Comment by fbrausse** [comment]
> > I don't understand why I failed the `PR / Check C/C++ code-style (pull_request)` test. Could someone please help me with this?
> 
> I think this is happening because the CI is trying to add a commit to fix the indentation in the 'inheritance_fix' branch in this repository, but the branch only exists in your fork (ChenfengWei0).

We should disable the auto-commit feature for non-local branches. Personally, I'd also like to disable it for branch names starting with fb/.

File: src/solidity-frontend/solidity_convert.h [hunk 1]
```diff
@@ -188,6 +168,38 @@ class solidity_convertert
   void convert_type_expr(const namespacet &ns, exprt &dest, const typet &type);
   bool convert_hex_literal(std::string the_value, exprt &dest, const int n = 0);
 
+  contextt &context;
+  namespacet ns;
+  nlohmann::json
+    &ast_json; // json for Solidity AST. Use vector for multiple contracts
```
Comment (fbrausse) [inline src/solidity-frontend/solidity_convert.h]: Please put the comment above the declaration, so it isn't broken across lines.

**Review by fbrausse — COMMENTED** [review]
In general looks good to me. One remark below, and one question here: Should the solidity symbols really be prefixed with `c:@`?

File: src/solidity-frontend/solidity_convert.h (hunk 1 above)
Comment (ChenfengWei0) [reply src/solidity-frontend/solidity_convert.h]: Changed as suggested, thanks

**Comment by ChenfengWei0** [comment]
> In general looks good to me. One remark below, and one question here: Should the solidity symbols really be prefixed with `c:@`?

Changed the prefix from `c:@` to `sol:@`.

**Comment by ChenfengWei0** [comment]
> > I don't understand why I failed the `PR / Check C/C++ code-style (pull_request)` test. Could someone please help me with this?
> 
> I think this is happening because the CI is trying to add a commit to fix the indentation in the 'inheritance_fix' branch in this repository, but the branch only exists in your fork (ChenfengWei0).

Thank you very much for your help, @brcfarias! It is indeed caused by the code format issue.

**Review by lucasccordeiro — APPROVED** [review]

**Review by fbrausse — APPROVED** [review]
Thanks @ChenfengWei0, LGTM.

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**